src/m365/commands/request.ts

```diff
@@ -0,0 +1,163 @@
+import { AxiosRequestConfig, AxiosRequestHeaders } from 'axios';
+import auth, { Auth } from '../../Auth';
+import { Logger } from '../../cli';
+import Command from '../../Command';
+import GlobalOptions from '../../GlobalOptions';
+import request from '../../request';
+import commands from './commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  url: string;
+  method?: string;
+  resource?: string;
+  body?: string;
+  headers?: string;
+  accept?: string;
+  contentType?: string;
+}
+
+class RequestCommand extends Command {
+  public get name(): string {
+    return commands.REQUEST;
+  }
+
+  public get description(): string {
+    return 'Invoke a custom request at a Microsoft 365 API';
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initValidators();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        method: args.options.method || 'get',
+        resource: args.options.resource,
+        accept: args.options.accept || 'application/json',
+        contentType: args.options.contentType,
+        body: typeof args.options.body !== 'undefined',
+        headers: typeof args.options.headers !== 'undefined'
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-u, --url <url>'
+      },
+      {
+        option: '-m, --method [method]',
+        autocomplete: ['get', 'post', 'put', 'patch', 'delete', 'head', 'options']
+      },
+      {
+        option: '-r, --resource [resource]'
+      },
+      {
+        option: '-b, --body [body]'
+      },
+      {
+        option: '-h, --headers [headers]'
+      },
+      {
+        option: '-a, --accept [accept]'
+      },
+      {
+        option: '-c, --contentType [contentType]'
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
```

Comment:
Since we only allow a predefined set of methods, we should validate if the specified method is allowed

Reply:
🤙